ocaml/lambda/value_rec_compiler.ml

```diff
@@ -327,23 +330,55 @@ let compute_static_size lam =
     | Popaque _
     | Pdls_get ->
         dynamic_size ()
-    | ( Pobj_dup | Parray_to_iarray | Parray_of_iarray | Pgetpredef _
-      | Pmakefloatblock (_, _) | Pmakeufloatblock (_, _) | Pufloatfield (_, _)
-      | Psetufloatfield (_, _) | Pmake_unboxed_product _
-      | Punboxed_product_field (_, _) | Punboxed_float_comp (_, _)
-      | Punboxed_int_comp (_, _) | Pstring_load_128 _
-      | Pbytes_load_128 _ | Pbytes_set_128 _ | Pbigstring_load_128 _
-      | Pbigstring_set_128 _ | Pfloatarray_load_128 _ | Pfloat_array_load_128 _
-      | Pint_array_load_128 _ | Punboxed_float_array_load_128 _
-      | Punboxed_int32_array_load_128 _ | Punboxed_int64_array_load_128 _
-      | Punboxed_nativeint_array_load_128 _ | Pfloatarray_set_128 _
-      | Pfloat_array_set_128 _ | Pint_array_set_128 _
-      | Punboxed_float_array_set_128 _ | Punboxed_int32_array_set_128 _
-      | Punboxed_int64_array_set_128 _ | Punboxed_nativeint_array_set_128 _
-      | Pprobe_is_enabled _ | Pobj_magic _ | Punbox_float _|Pbox_float (_, _)
-      | Punbox_int _ | Pbox_int (_, _) | Pget_header _) ->
-        (* New primitives -- need to check *)
-        Misc.fatal_error "size_of_primitives: unhandled primitive"
+
+    (* Note: the primitives below are not present upstream,
+       so they haven't been reviewed *)
+    | Pmakefloatblock (_, _) ->
+        let size = List.length args in
+        Block (Float_record size)
+
+    | Psetufloatfield (_, _)
+    | Pbytes_set_128 _
+    | Pbigstring_set_128 _
+    | Pfloatarray_set_128 _
+    | Pfloat_array_set_128 _
+    | Pint_array_set_128 _
+    | Punboxed_float_array_set_128 _
+    | Punboxed_int32_array_set_128 _
+    | Punboxed_int64_array_set_128 _
+    | Punboxed_nativeint_array_set_128 _ ->
+        Constant
+
+    | Pmakeufloatblock (_, _)
+    | Pmake_unboxed_product _ ->
+        dynamic_size () (* Not allowed *)
```

Comment:
`dynamic_size` calls `fatal_error`. Unless you think this particular case is more likely to happen than the others, I don't think it's worth diverging. The comment means that we could pre-allocate (if we added the right primitives), but since we don't allow it during typing here we can reject it.

Reply:
Oops. I was wondering in the back of my head what we could do in the dynamic case, but never bothered to look at the definition. Resolving.